src/drawer.canvasentry.js

```diff
@@ -146,14 +146,25 @@ export default class CanvasEntry {
     /**
      * Set the fill styles for wave and progress
      *
-     * @param {string} waveColor Fill color for the wave canvas
-     * @param {?string} progressColor Fill color for the progress canvas
+     * @param {string|string[]} waveColor Fill color for the wave canvas, or an array of colors to apply as a gradient
+     * @param {?string|string[]} progressColor Fill color for the progress canvas, or an array of colors to apply as a gradient
      */
     setFillStyles(waveColor, progressColor) {
-        this.waveCtx.fillStyle = waveColor;
-
+        if (typeof waveColor == 'string') {
+            this.waveCtx.fillStyle = waveColor;
+        } else if (typeof waveColor == 'object') {
+            const waveGradient = this.waveCtx.createLinearGradient(0, 0, 0, this.waveCtx.canvas.height);
+            waveColor.forEach((color, index) => waveGradient.addColorStop((index / waveColor.length), color));
+            this.waveCtx.fillStyle = waveGradient;
+        }
         if (this.hasProgressCanvas) {
-            this.progressCtx.fillStyle = progressColor;
+            if (typeof progressColor == 'string') {
```

Comment:
it looks like a duplicate of previous code, I'd rather extract it to an utility or something like this:
```
this.waveCtx.fillStyle = getFillStyle(this.waveCtx, waveColor)

if (this.hasProgressCanvas) {
  this.progressCtx.fillStyle = getFillStyle(this.progressCtx, progressColor)
}


...

export const getFillStyle = (context, color) => {
  if (typeof waveColor == 'string') {
    return color
  }

  const waveGradient = context.createLinearGradient(0, 0, 0, context.canvas.height);
  color.forEach((value, index) => waveGradient.addColorStop((index / color.length), value));

  return waveGradient
}
```

Reply:
Agreed, included in latest commit.